src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -86,52 +88,65 @@ private static unsafe ulong GetMaskAsLong(byte[] bytes)
         }
 
         // The same as GetAsciiStringNonNullCharacters but throws BadRequest
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe string GetHeaderName(this ReadOnlySpan<byte> span)
         {
             if (span.IsEmpty)
             {
                 return string.Empty;
             }
 
-            var asciiString = new string('\0', span.Length);
+            fixed (byte* source = &MemoryMarshal.GetReference(span))
+            {
+                return string.Create(span.Length, new IntPtr(source), s_getHeaderName);
+            }
+        }
 
-            fixed (char* output = asciiString)
-            fixed (byte* buffer = span)
+        private static readonly SpanAction<char, IntPtr> s_getHeaderName = GetHeaderName;
```

Comment:
Nit: can we move the static field to the top of the class and start it with `_` for consistency? I know the convention is different in the runtime repo.

Reply:
Done -- will be updated with the next push.